bytecomp/bytelink.ml

```diff
@@ -622,16 +627,13 @@ let link objfiles output_name =
     | _                  -> "stdlib.cma" :: objfiles @ ["std_exit.cmo"]
   in
   let tolink = List.fold_right scan_file objfiles [] in
-  let missing_modules =
-    Ident.Map.filter (fun id _ -> not (Ident.is_predef id)) !missing_globals
-  in
   begin
-    match Ident.Map.bindings missing_modules with
+    match Idents.Map.bindings !missing_globals with
     | [] -> ()
-    | (id, cu_name) :: _ ->
+    | (Global id, cu_name) :: _ ->
         match !badly_ordered_dependencies with
         | [] ->
-            raise (Error (Required_module_unavailable (Ident.name id, cu_name)))
+            raise (Error (Required_module_unavailable (id, cu_name)))
```

Comment:
You could change the type of `Required_module_unavailable` to use global instead of string.